pkgs/development/tools/ansible-language-server/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+}:
+
+buildNpmPackage rec {
+  pname = "ansible-language-server";
+  version = "1.0.2-next.0";
+
+  src = fetchFromGitHub {
+    owner = "ansible";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-E4xWwqXl5n/eChJ8JM32K2gTYE/F8Y76J3Sll++48Uo=";
+  };
+
+  npmDepsHash = "sha256-8FP6hF85w1Zbhiwi2V350ZWFAykAfvsXRGL8bvGk1XE=";
+  npmBuildScript = "compile";
+
+  postPatch = ''
+    sed -i '/"prepare"/d' package.json
+    sed -i '/"prepack"/d' package.json
```

Comment:
Perhaps we should write why these are required? (I can, with detail, if you want.)

Reply:
Feel free to recommend something.

Reply:
How about this?

> We remove the `prepare` and `prepack` scripts because they run the build script, and therefore are redundant.
>
> Additionally, the `prepack` script runs `npm ci` in addition to the build script. Directly before `npm pack` is run, we make npm unaware of the dependency cache, causing the `npm ci` invocation to fail, wiping out `node_modules`, which causes a mysterious error stating that `tsc` isn't installed.

Or is this too unwieldy?